com.unity.multiplayer.mlapi/Tests/Runtime/MultiInstance/MultiInstanceHelpers.cs

```diff
@@ -183,23 +270,56 @@ public static IEnumerator WaitForClientConnectedToServer(NetworkManager server,
                 throw new InvalidOperationException("Cannot wait for connected as client");
             }
 
-            int startFrame = Time.frameCount;
+            var startFrame = Time.frameCount;
 
             while (Time.frameCount - startFrame <= maxFrames && server.ConnectedClients.Count != (server.IsHost ? 2 : 1))
             {
-                int nextFrameId = Time.frameCount + 1;
-                yield return new WaitUntil(() => Time.frameCount >= nextFrameId);
+                var nextFrameNumber = Time.frameCount + 1;
+                yield return new WaitUntil(() => Time.frameCount >= nextFrameNumber);
+            }
+
+            var res = server.ConnectedClients.Count == (server.IsHost ? 2 : 1);
+
+            if (result != null)
+            {
+                result.Result = res;
+            }
+            else
+            {
+                Assert.True(res, "A Client never connected to server");
+            }
+        }
+
+        /// <summary>
+        /// Waits on the server side for 1 client to be connected
+        /// </summary>
+        /// <param name="server">The server</param>
+        /// <param name="result">The result. If null, it will automatically assert</param>
+        /// <param name="maxFrames">The max frames to wait for</param>
+        public static IEnumerator WaitForClientsConnectedToServer(NetworkManager server, int clientCount, CoroutineResultWrapper<bool> result = null, int maxFrames = 64)
```

Comment:
Why do we have a duplicate method for this? Why not just have a default clientCount=1?

Reply:
Good point, I just made some adjustments which will remove the duplicate code while still keeping the methods in-tact.
( _Albin was the original author so I will leave it up to him to determine if we should completely remove the "wrapper methods"_)